Source/BCfill.cpp

```diff
@@ -21,14 +21,18 @@ struct PCHypFillExtDir
   void operator()(
     const amrex::IntVect& iv,
     amrex::Array4<amrex::Real> const& dest,
-    const int /*dcomp*/,
-    const int /*numcomp*/,
+    const int dcomp,
+    const int numcomp,
     amrex::GeometryData const& geom,
     const amrex::Real time,
     const amrex::BCRec* bcr,
     const int /*bcomp*/,
     const int /*orig_comp*/) const
   {
+    // We need the whole state to apply physical BCs
+    AMREX_ASSERT(dcomp == 0 and numcomp == NVAR);
```

Comment:
you could switch this to an `AMREX_ALWAYS_ASSERT` and then you don't need the ignore_unused. Might be good to always assert this one. 

Also, can you use `&&` and not `and` ?